script.js

```diff
@@ -1,61 +1,66 @@
+class BookList {
+  constructor() {
+    this.books = JSON.parse(localStorage.getItem('books')) || [];
+    this.bookList = document.getElementById('book-list');
+    this.titleInput = document.getElementById('title');
+    this.authorInput = document.getElementById('author');
 
-// Load the books from localStorage or use an empty array
-const books = JSON.parse(localStorage.getItem('books')) || [];
-
-function displayBooks() {
-  // Clear the current book list
-  bookList.innerHTML = '';
-
-  // Add each book to the list
-  books.forEach((book, index) => {
-    const li = document.createElement('li');
-    li.innerText = `${book.title} by ${book.author}`;
-    bookList.appendChild(li);
+    document.getElementById('book-form').addEventListener('submit', (e) => {
+      e.preventDefault();
+      this.addBook();
+    });
 
-    // Add a remove button to each book
-    const removeButton = document.createElement('button');
-    removeButton.innerText = 'Remove';
-    li.appendChild(removeButton);
+    this.displayBooks();
+  }
 
-    // Add an event listener to the remove button
-    removeButton.addEventListener('click', () => {
-      // Remove the book from the collection
-      books.splice(index, 1);
+  addBook() {
+    const book = {
+      title: this.titleInput.value,
+      author: this.authorInput.value,
+      remove: () => {
+        this.books = this.books.filter((b) => b.title !== book.title);
+        localStorage.setItem('books', JSON.stringify(this.books));
+        this.displayBooks();
+      },
+    };
 
-      // Save the updated book collection to localStorage
-      localStorage.setItem('books', JSON.stringify(books));
+    this.books.push(book);
+    localStorage.setItem('books', JSON.stringify(this.books));
+    this.displayBooks();
+    this.titleInput.value = '';
+    this.authorInput.value = '';
+  }
 
-      // Remove the book from the page
-      li.remove();
-    });
-  });
-}
+  displayBooks() {
+    this.bookList.innerHTML = '';
 
-function addBookToCollection(event) {
-  // Prevent the default form submission behavior
-  event.preventDefault();
+    this.books.forEach((book) => {
+      const li = document.createElement('li');
+      li.classList.add('book'); // Agregar la clase "book" al li
 
-  // Get the values of the form inputs
-  const title = document.getElementById('title-input').value;
-  const author = document.getElementById('author-input').value;
+      const title = document.createElement('span');
+      title.textContent = book.title;
+      li.appendChild(title);
 
-  // Add the new book to the collection
-  books.push({ title, author });
+      const author = document.createElement('span');
+      author.textContent = book.author;
+      li.appendChild(author);
 
-  // Clear the form inputs
-  addBookForm.reset();
+      const removeBtn = document.createElement('button');
+      removeBtn.textContent = 'Remove';
+      li.appendChild(removeBtn);
```

Comment:
[OPTIONAL] Well done overall 💯 but there seems to be some repetition here, in order to improve the code you could look to create a helper function here to avoid the repetition and respect the `DRY` principles. 🤓

Reply:
No sure why it is not getting updated once it was deleted from our end. I'll be pushing it again to solve it.  Thanks for your review Didier.